src/components/TrackListItem.js

```diff
@@ -0,0 +1,94 @@
+import PropTypes from "prop-types";
+import {
+    Container,
+    Row,
+    Col
+  } from "reactstrap";
+import {TrackFilePicker} from './TrackFilePicker';
+import {TrackTypeDropdown} from './TrackTypeDropdown';
+import {TrackDeleteButton} from './TrackDeleteButton';
+import {TrackSettingsButton} from './TrackSettingsButton';
+
+
+export const TrackListItem = ({
```

Comment:
There needs to be some documentation here for what `onChange` has to be written to expect. Otherwise it will be hard to figure out what you need to do to use the component.

Right now it looks like onChange needs to accept 3 arguments: a string track type, a string track filename, and an object of the color settings type, all in that order. It also looks like right now it needs to be able to cope with `undefined` values, but I want it to be protected from ever seeing them instead.

I guess if the track type, file, and color settings all come in as separate props, that makes sense.

If we ever switch to just one prop for the track object as a whole, we'd then want to switch to a function that takes a modified copy of the track object.